packages/nx-plugin/src/executors/verify-deps/executor.ts

```diff
@@ -0,0 +1,110 @@
+  const { projectName, projectsConfigurations, root: workspaceRoot } = context;
+  if (!projectsConfigurations || !projectName) {
+    throw new Error('no project configurations');
+  }
+
+  const { root } = projectsConfigurations.projects[projectName];
+  const { name: tmpDir, removeCallback: cleanup } = tmp.dirSync({
+    prefix: projectName,
+    unsafeCleanup: true,
+  });
+  console.info('copying', projectName, 'to', tmpDir);
+  copySync(root, tmpDir);
+  const paths = getPackagePaths(tmpDir, '.', true);
+
+  copyYarnBinary(paths, workspaceRoot);
+
+  configureTsConfig(paths, workspaceRoot);
+
+  configurePackageJson(paths, workspaceRoot);
+
+  let success = true;
+  try {
+    const yarnCmd = 'yarn';
+    console.log('Run', yarnCmd);
+    execSync(yarnCmd, { stdio: 'inherit', cwd: paths.root });
+
+    const tscCmd =
+      'npx -p typescript tsc --noEmit --project ./tsconfig.lib.json';
+    console.log('Run', tscCmd);
+    execSync(tscCmd, {
+      stdio: 'inherit',
+      cwd: paths.root,
+    });
+  } catch {
+    success = false;
+    console.error(
+      '======================================================================'
+    );
+    console.error(
+      'verify-deps check failed. Most likely the minimum version of your peer dependency ranges need to be updated!'
+    );
+    console.error('You can debug by opening the test project', tmpDir);
+    console.error(
+      'and run the command `npx -p typescript tsc --noEmit --project ./tsconfig.lib.json`'
+    );
+  }
+
+  if (shouldCleanup && success) {
+    cleanup();
+  }
+  return {
+    success,
+  };
+}
+
+function copyYarnBinary(paths: PackagePaths, workspaceRoot: string) {
+  copySync(path.join(workspaceRoot, '.yarn'), path.join(paths.root, '.yarn'));
+}
+
+function configureTsConfig(paths: PackagePaths, workspaceRoot: string) {
+  copySync(
+    path.join(workspaceRoot, 'tsconfig.base.json'),
+    path.join(paths.root, 'tsconfig.base.json')
+  );
+
+  const tsconfig = readJsonFile(paths.tsconfig);
+  tsconfig.extends = './tsconfig.base.json';
+  writeJsonFile(paths.tsconfig, tsconfig);
+}
+
+function configurePackageJson(paths: PackagePaths, workspaceRoot: string) {
+  const pkgJson = readJsonFile(paths.packageJson);
+  pkgJson.dependencies ??= {};
+  const peerDeps = Object.entries(pkgJson.peerDependencies) as Array<
+    [string, string]
+  >;
+  // use minimum version of peer deps to test
+  for (const [depName, versionRange] of peerDeps) {
+    const minVersion = semver.minVersion(versionRange);
+    if (minVersion) {
+      pkgJson.dependencies[depName] = minVersion.version;
+    }
+  }
+
+  // Copy all types dependencies - they are not that big and there's not many of them (yet)
+  const rootPkgJson = readJsonFile(path.join(workspaceRoot, 'package.json'));
+  const rootDevDeps = Object.entries(rootPkgJson.devDependencies ?? {});
+  pkgJson.devDependencies ??= {};
+  for (const [depName, version] of rootDevDeps) {
+    if (depName.startsWith('@types/')) {
```

Comment:
This will also copy these, correct?

https://github.com/microsoft/fluentui-contrib/blob/823224320f87b291d794c0e18da4de9c4b5cf5a0/package.json#L45-L47

Reply:
yeah, there's no real way to find out what types pacakges are necessary without doing an AST parse of the src

Reply:
I would rather do just `@types/react` & `@types/react-dom` instead. This should cover all use cases.